beetsplug/fish.py

```diff
@@ -89,14 +95,22 @@ def run(self, lib, opts, args):
         # If specified, also collect the values for these fields.
         # Make a giant string of all the above, formatted in a way that
         # allows Fish to do tab completion for the `beet` command.
-        home_dir = os.path.expanduser("~")
-        completion_dir = os.path.join(home_dir, '.config/fish/completions')
-        try:
-            os.makedirs(completion_dir)
-        except OSError:
-            if not os.path.isdir(completion_dir):
-                raise
-        completion_file_path = os.path.join(completion_dir, 'beet.fish')
+
+        completion_file_path = opts.output
+        if completion_file_path is not None:
+            completion_dir = os.path.dirname(completion_file_path)
+        else:
+            home_dir = os.path.expanduser("~")
+            completion_dir = os.path.join(home_dir, '.config/fish/completions')
+            completion_file_path = os.path.join(completion_dir, 'beet.fish')
+
+        if completion_dir != '':
+            try:
+                os.makedirs(completion_dir)
+            except OSError:
+                if not os.path.isdir(completion_dir):
+                    raise
```

Comment:
I know this is not new, but I think we should probably just do `makedirs(completion_dir, exist_ok=True)`, which avoids the need for the exception handler.

Reply:
Done.